vpc/service/branch_enis.go

```diff
@@ -1404,3 +1406,45 @@ func (vpcService *vpcService) createBranchENI(ctx context.Context, tx *sql.Tx, s
 
 	return iface, nil
 }
+
+func (vpcService *vpcService) ListTrunkToBranchENIMappings(ctx context.Context, req *vpcapi.ListTrunkToBranchENIMappingsRequest) (*vpcapi.ListTrunkToBranchENIMappingsResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "listTrunkToBranchENIMappings")
+	defer span.End()
+
+	mappings := []*vpcapi.TrunkENIMapping{}
+
+	tx, err := vpcService.db.BeginTx(ctx, &sql.TxOptions{
+		ReadOnly: true,
+	})
+	if err != nil {
+		err = errors.Wrap(err, "Cannot start database transaction")
+		tracehelpers.SetStatus(err, span)
+		return nil, err
+	}
+	rows, err := tx.QueryContext(ctx, `
```

Comment:
Aren't you leaking the transaction here? You need to do defer tx.rollback, or tx.commit.

Reply:
Hmm, `mock.ExpectRollback()` doesn't seem to complain, even though I don't have those.

For this API, do I even really need a transaction?

Reply:
I think you should have a txn to indicate this can be a read only snapshot query.

Reply:
It is currently read only, adding snapshot level isloation gives me:
`"Cannot start database transaction: pq: isolation level not supported: 5"`

How about RepeatableRead ?

Reply:
err, When I meant snapshot, I didn't mean snapshot isolation, I meant that it was a read only query, so postgres could handle it as using a consistent cut / snapshot

Reply:
Ok I think I understand, the the fact that it is in a tx and is readonly is enough.